src/main.py

```diff
@@ -20,19 +22,47 @@
 # Middleware to automatically open/close a database connection for every request
 @app.middleware('request')
 async def handle_request(request):
+    """Open database connection before each request is handled."""
```

Comment:
I personally think a # comment is a bit more readable than docstrings for single-line comments, but I don't know what PEP8 says

Reply:
Apparently docstring is the preferred method and can be access directly in Python using the __doc__